### ISSUE TITLE
Huge increase of "high risk requests" out of nowhere

### ISSUE THREAD

**Issue by pierroo** [douglasjunior/react-native-recaptcha-that-works#67]
- What version of React Native? 0.73.4
- What version of React Native WebView? 13.7.0
- What version of the library? 1.3.0
- iOS/Android version?
- Did the problem happen after updating React Native? no
- Are you using the library for the first time? no

I have been using this library for years without issue.
Then out of nowhere starting November 15th approximately, the "high risk requests" went from a usual 0.5% to litterally 75%!!
which blocks almost everyone from signing up in my app.

What could explain this issue?
any help would be super appreciated

**Comment by adamhongmy** [comment]
My company app is facing same spike issue starting on 16th November sgt. But we are using react-native-webview. I am now searching around and found this github issue and wanted to share my issue too.

**Comment by adamhongmy** [comment]
- What version of React Native? 0.73.4
- What version of React Native WebView?  0.74.6
- recapatcha version? v3

**Comment by adamhongmy** [comment]
I am wondering are there any google forum post that reported this issue.

**Comment by pierroo** [comment]
> * What version of React Native? 0.73.4
> * What version of React Native WebView?  0.74.6
> * recapatcha version? v3

ok so upgrading recaptcha won't fix the issue;
you're on the same RN version as me
your webview version is probably wrong though, are you using "react-native-webview" as well?

as for google forum post, didn't find any yet, but first one who finds anything shares it here alright?

**Comment by adamhongmy** [comment]
` "react-native-webview": "11.26.1",` 
the version is incorrect? how so? 😮

**Comment by adamhongmy** [comment]
ya i am using ` "react-native-webview": "11.26.1",`. I corrected the comment.

**Comment by douglasjunior** [comment]
Hello guys, thank you for the feedback.

I'm currently not working on a project that uses reCAPTCHA, so I'm unsure about the blocks. However, I will provide any updates if I have them.

In the past, the blocks I encountered were related to multiple devices solving captchas using the same internet connection (Wi-Fi), which led Google to add the IP address to a sort of "blacklist."

_Just for the record reCAPTCHA v2 and v3 are completely different products._

**Comment by pierroo** [comment]
thank you for your input @douglasjunior 

but in my case, it's thousands of users everyday from every parts of the world that faces the issue.
I have like 5000 daily new subscription that goes through this captcha, and 50% of them are failing because of a score too low.

so that can't be the IP blacklist issue, when they email asking for help they are indeed genuine users

**Comment by shriganesh64** [comment]
+1. I have been using this package for approximately 1.5 years and updated react-native-webview to version 13.12.1 three months ago. However, this issue started occurring about a week ago. Our company operates white-labeled food ordering applications, and this problem is significantly impacting user sign-ins across multiple merchants.

A workaround or solution would be greatly appreciated as this is causing disruptions for many users.

**Comment by douglasjunior** [comment]
According to the latest reports on this issue, the problem started without anything being changed in the lib or webview.

This makes me think that some changes on Google's side increased the rejection rate.

Therefore, I don't see anything we can do in relation to the lib to solve this.

**Comment by pierroo** [comment]
You're right @douglasjunior , in that case does any of you know which Google's service / contact we should reach out to to let them know about it?
if we all contact them for the same it might make them realize something legit is going on

**Comment by lvillacin** [comment]
Encountering this issue on our end too. I was thinking in a normal website, the reCAPTCHA script is added as soon as the page is loaded and reCAPTCHA may already start gathering information about the device / user.

When used in react native inside a web view however, we load the reCAPTCHA script just in that moment (few seconds max) we're generating the token.

I was thinking if there's a way for us to move the reCAPTCHA script on screen load and perhaps keeping it there without generating a token and then just call the `grecaptcha.execute()` function later on when user submits the button may improve our stats?

We haven't tried it yet and would love some input on this.

**Comment by JRS-Developer** [comment]
Hello, is this issue still happening? I was thinking on using this library migrating from Recaptcha v3 (low score) and I'm not sure now.

**Comment by pierroo** [comment]
@JRS-Developer still happening yes;
However not sure this is related to this library specifically; but to captcha in general.
Are you not facing it with Recaptcha v3?

**Comment by JRS-Developer** [comment]
@pierroo We don't have much traffic so I can't say if recaptcha v3 works better or no. Right now our issue with it is that it has low scores in mobile devices and users need to submit forms two times to make it work.

**Comment by adamhongmy** [comment]
Hi, 
I wanted to provide an update on my findings. My team took the time to investigate and review the official documentations. 

One of the [document](https://services.google.com/fh/files/blogs/recaptcha_enterprise_guidebook.pdf) mentioned mentions the use of WebView:

> Please note that for iOS and Android, Webview based implementation is not currently recommended.

![Image](https://github.com/user-attachments/assets/8b95592b-578d-482e-a18d-15823bbd0efb)

Digging deeper, the document is suggesting to use their [first party solution for React Native](https://github.com/GoogleCloudPlatform/recaptcha-enterprise-react-native/tree/main?tab=readme-ov-file) but it is intended for enterprise use.

After discussing with my team, we decided to try it out by setting up the enterprise enviroment 
Then, I created 2 sample apps: Webview and Native (aka first-party library).

I ran both apps on an actual Android device, and here are the results:

- Webview score: 0.1
- Native score: 0.8999999761581421

I will report this to my team tomorrow to determine our next steps based on this information.

Hope you guys find this somewhat useful.

**Comment by pierroo** [comment]
Thank you @adamhongmy for taking the time to share your findings!
So the summary is: once again Google is making a di*k move and probably forcing their paid solution by sneakily disabling other free alternatives.
I am no native expert, but could this library be adjusted to accomodate a native solution?

**Comment by douglasjunior** [comment]
Thank you for your findings @adamhongmy, I will update the Readme with this information.

**Comment by tanishc-bm-ai** [comment]
@pierroo @douglasjunior let's say If I am going for a Standard or Enterprise model of recaptcha, how does it make any difference whether I use this package or the one provided by google.

I just need help in choosing what should I use if I am going to have more than 300k requests per month for my login requests.
